src/Pate/Memory/MemTrace.hs

```diff
@@ -1575,13 +1630,31 @@ instance PEM.ExprMappable sym (MemOp sym w) where
 instance PEM.ExprMappable sym (MemEvent sym w) where
   mapExpr sym f = \case
     MemOpEvent op -> MemOpEvent <$> PEM.mapExpr sym f op
-    SyscallEvent arg -> SyscallEvent <$> f arg
+    SyscallEvent i arg -> SyscallEvent i <$> f arg -- TODO? rewrite the mux tree?
+
+instance PEM.ExprMappable sym a => PEM.ExprMappable sym (SymSequence sym a) where
+  mapExpr sym f = evalWithFreshCache $ \rec -> \case
+    SymSequenceNil -> nilSymSequence sym
+    SymSequenceCons _ x xs ->
+      do x'  <- PEM.mapExpr sym f x
+         xs' <- rec xs
+         consSymSequence sym x' xs'
+    SymSequenceAppend _ xs ys ->
+     do xs' <- rec xs
+        ys' <- rec ys
+        appendSymSequence sym xs' ys'
+    SymSequenceMerge _ p xs ys ->
+     do p' <- f p
+        iteM muxSymSequence sym p' (rec xs) (rec ys)
```

Comment:
This turned out to be important to avoid spurious counterexamples involving observables. I think we can avoid needing this (and the whole ExprMappable rewriting infrastructure) if we are a bit more carful about how we set up the initial states.